### ISSUE TITLE
deprecateTypingAliases incorrectly reports Optional as deprecated

### ISSUE THREAD

**Issue by roel-gerrits** [DetachHead/basedpyright#1718]
### Description

``` python
from typing import Optional
```
... causes deprecation warning, but it should not.

Playground example: https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDyCMSYKAhgDZA

This problem also occurs in pyright: https://github.com/microsoft/pyright/issues/9793

There is a bit of discussion in the pyright issue, I think [smuuf's comment ](https://github.com/microsoft/pyright/issues/9793#issuecomment-2766343994)has the right conclusion.

Basically: Basedpyright should not report something as deprecated if it is not [officially deprecated](https://docs.python.org/3/library/typing.html#deprecated-aliases).

Edit: 
I now see that this has been reported before. I've mostly read those discussions. The thing that I find problematic is that although there might be good reasons to use `T | None` instead of `Optional[T]`, **it is not deprecated**!'

So either this rule should result in a different warning, or it should be removed.

Reporting it as deprecated just causes lots of confusion, as people think: "Oh shit am I using a deprecated function, let's see what the docs say about it, and find nothing, questioning there sanity."

**Comment by DetachHead** [comment]
see my comment at https://github.com/DetachHead/basedpyright/issues/1682#issuecomment-3676952903

**Comment by roel-gerrits** [comment]
Thanks for linking that comment. 

I think the point left undiscussed is the confusion that is generated by this warning. I think it's a good thing if users can trust there tooling to tell the truth, and in the end it's up to the Python maintainers to declare something deprecated. 

Of course it's nice that a tool (like basedpyright) informs the user about possible improvements to there code. In my opinion this specific rule should not result in a deprecation warning, but instead some type of hint, it should therefore also not be grouped behind the `deprecateTypingAliases` switch, because the other typing aliasses (`List`, `Set`, etc) behind that switch **_are_** actually deprecated.

**Comment by DetachHead** [comment]
i understand your point. however i try to keep basedpyright backwards compatible with pyright as much as possible. this means for the most part i keep the behavior of existing rules intact and implement new functionality using new rules so that users can easily turn them off if they want basedpyright's behavior to match pyright's.

this means if i were to make this change, it would involve marking the `deprecateTypingAliases` setting as ["discouraged"](https://docs.basedpyright.com/latest/configuration/config-files/#discouraged-settings_1) and creating two new versions of the setting just to make the distinction between types that have been "officially" and "implicitly" deprecated.

i just don't see the point in doing that when users who want more control over this rule already have the option to use ruff.

i guess the number of times this has been raised is enough evidence that this situation is confusing to users. i'd be open to changing the error message for `Optional` to say something like:

> While `Optional` is not specifically deprecated, it's recommended to use `int | None` instead

we can probably also add some info about this to the docs, linking to [this comment](https://discuss.python.org/t/clarification-for-pep-604-is-foo-int-none-to-replace-all-use-of-foo-optional-int/26945/6) (which IMO is the most solid argument to avoid using `Optional`) as well as explaining how users can configure the ruff rules to their liking if they choose to continue using it

**Comment by roel-gerrits** [comment]
I think keeping in line with pyright is a good strategy, but introducing  introducing new settings will be probably annoying to maintain. I could try and create a issue on the pyright repo. 

Meanwhile changing the error message and update the docs with the linked comment would be super useful, and reduce confusion (and more github issues 😉).
